openstack/messaging/v2/messages/requests.go

```diff
@@ -0,0 +1,54 @@
+package messages
+
+import (
+	"github.com/gophercloud/gophercloud"
+)
+
+// CreateOptsBuilder Builder.
+type CreateOptsBuilder interface {
+	ToMessageCreateMap() (map[string]interface{}, error)
+}
+
+// Messages params to be used with CreateOpts.
+type Messages struct {
+	// TTL specifies how long the server waits before marking the message
+	// as expired and removing it from the queue.
+	TTL int `json:"ttl,omitempty"`
+
+	// Delay specifies how long the message can be claimed.
+	Delay int `json:"delay,omitempty"`
+
+	// Body specifies an arbitrary document that constitutes the body of the message being sent.
+	Body map[string]interface{} `json:"body" required:"true"`
+}
+
+// CreateOpts params to be used with Create.
+type CreateOpts struct {
+	// QueueName specifies the name of the queue to create messages on.
+	QueueName string `json:"queue_name" required:"true"`
```

Comment:
Totally up to you on this one:

Since there's a hard relationship between a message and a queue, _and_ the queue name is part of the URL, the queue name could go in the function signature:

```go
Create(client, queueName, createOpts)
```

The main reason I didn't like this for Queues was because there wasn't such a relationship.

If you wanted to make queue name part of the function, then the only remaining field in `CreateOpts` would be `Messages`. In that case, you could change this to the Batch pattern like in the queues update.